test/functional/api-demos/ime-e2e-specs.js

```diff
@@ -0,0 +1,50 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { AndroidDriver } from '../../..';
+import sampleApps from 'sample-apps';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+let driver;
+let defaultCaps = {
+  app: sampleApps('ApiDemos-debug'),
+  deviceName: 'Android',
+  platformName: 'Android',
+  appActivity: "view.Controls1",
+  unicodeKeyboard: true,
+  resetKeyboard: true
+};
+let unicodeImeId = 'io.appium.android.ime/.UnicodeIME';
+
+describe('apidemo - IME', function () {
+  before(async () => {
+    driver = new AndroidDriver();
+    await driver.createSession(defaultCaps);
+  });
+  beforeEach(async () => {
+    await driver.reset();
```

Comment:
`reset` seems pretty heavy. I think it might be possible to just use `launchActivity`. Throughout.